lib/spandex_datadog/api_server.ex

```diff
@@ -95,12 +96,28 @@ defmodule SpandexDatadog.ApiServer do
       waiting_traces: [],
       batch_size: opts[:batch_size],
       sync_threshold: opts[:sync_threshold],
-      agent_pid: agent_pid
+      agent_pid: agent_pid,
+      container_id: get_container_id()
     }
 
     {:ok, state}
   end
 
+  @cgroup_uuid "[0-9a-f]{8}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{12}"
+  @cgroup_ctnr "[0-9a-f]{64}"
+  @cgroup_task "[0-9a-f]{32}-\\d+"
+  @cgroup_regex Regex.compile!("/(#{@cgroup_uuid}|#{@cgroup_ctnr}|#{@cgroup_task})(?:.scope)?$", "m")
+
+  defp get_container_id() do
+    with {:ok, file_binary} <- File.read("/proc/self/cgroup"),
+         true <- String.valid?(file_binary),
```

Comment:
I'm not sure this is needed - `Regex` works with binaries, and the binary either matches the pattern or not. Doesn't really matter if the string contains valid characters as far as I understand.